microprofile/grpc/client/src/main/java/io/helidon/microprofile/grpc/client/model/ClientServiceModeller.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.microprofile.grpc.client.model;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import io.helidon.grpc.client.ClientMethodDescriptor;
+import io.helidon.grpc.client.ClientServiceDescriptor;
+import io.helidon.grpc.core.MethodHandler;
+import io.helidon.microprofile.grpc.core.GrpcMarshaller;
+import io.helidon.microprofile.grpc.core.RpcMethod;
+import io.helidon.microprofile.grpc.core.model.AbstractServiceModeller;
+import io.helidon.microprofile.grpc.core.model.AnnotatedMethod;
+import io.helidon.microprofile.grpc.core.model.AnnotatedMethodList;
+import io.helidon.microprofile.grpc.core.model.Instance;
+import io.helidon.microprofile.grpc.core.model.ModelHelper;
+
+/**
+ * Utility class for constructing a {@link ClientServiceDescriptor.Builder}
+ * from an annotated POJO.
+ */
+public class ClientServiceModeller
```

Comment:
Is this similar to MP REST Client? If so, then please have a look at the naming used in REST client and use similar approach.

Reply:
No I don't think that it is like the REST client  it is more like a builder. I have refactored it to be a Builder.